src/stactools/datacube/stac.py

```diff
@@ -76,6 +76,32 @@ def get_time_offset_and_step(unit: str) -> Tuple[datetime, timedelta]:
     raise ValueError(f"Failed to parse time unit from '{unit}'")
 
 
+def iso_duration(td: timedelta) -> str:
+    date_intervals = [
+        ('W', 604800),  # 60 * 60 * 24 * 7
+        ('D', 86400),    # 60 * 60 * 24
+    ]
+    time_intervals = [
+        ('H', 3600),    # 60 * 60
+        ('M', 60),
+    ]
+    seconds = td.total_seconds()
+    result = ["P"]
+    for name, count in date_intervals:
+        value = int(seconds // count)
+        if value:
+            seconds -= value * count
+            result.append(f"{value}{name}")
+    result.append("T")
```

Comment:
If we're e.g. exactly one day, we won't need the `T` right?

Reply:
That is correct. Will revise

Reply:
Done